pyvista/utilities/errors.py

```diff
@@ -21,10 +25,74 @@ def set_error_output_file(filename):
     return fileOutputWindow, outputWindow
 
 
+class VtkErrorCatcher:
+    """Context manager to temporarily catch VTK errors.
+
+    Parameters
+    ----------
+    raise_errors : bool, optional
+        Raise a ``RuntimeError`` when a VTK error is encountered.  Defaults to ``False``.
+
+    send_to_logging : bool, optional
+        Determine whether VTK errors raised within the context should also be sent to logging.  Defaults to ``True``.
+
+    Examples
+    --------
+    Catch VTK errors using the context manager.
+
+    >>> import pyvista
+    >>> error_catcher = pyvista.utilities.VtkErrorCatcher()
+    >>> with error_catcher:
+    ...     sphere = pyvista.Sphere()
+    """
+
+    def __init__(self, raise_errors=False, send_to_logging=True):
+        """Initialize context manager."""
+        self.raise_errors = raise_errors
+        self.send_to_logging = send_to_logging
+
+    def __enter__(self):
+        """Observe VTK string output window for errors."""
+        error_output = _vtk.vtkStringOutputWindow()
+        error_win = _vtk.vtkOutputWindow()
+        self._error_output_orig = error_win.GetInstance()
+        error_win.SetInstance(error_output)
+        obs = Observer(log=self.send_to_logging, store_history=True)
+        obs.observe(error_output)
+        self._observer = obs
+
+    def __exit__(self, type, val, traceback):
+        """Stop observing VTK string output window."""
+        error_win = _vtk.vtkOutputWindow()
+        error_win.SetInstance(self._error_output_orig)
+        self.events = self._observer.event_history
+        if self.raise_errors and self.events:
+            errors = [RuntimeError(f'{e.kind}: {e.alert}', e.path, e.address) for e in self.events]
+            raise RuntimeError(errors)
+
+
+@contextlib.contextmanager
+def raise_vtk_errors():
+    """Context manager to temporarily raise VTK errors.
```

Comment:
I think it may be confusing to users to have more than one context manager option.  It took me a careful read of the parameters and descriptions to get it.  I think either a short guide in one or both docstrings on what the difference is, or collapsing into a single context manager, would be helpful.

Reply:
I'd also prefer having a single context manager.  @whophil, do you think this is possible?

Reply:
`raise_vtk_errors` is just a convenience method which wraps  `VtkErrorCatcher`. However, you can use `VtkErrorCatcher` directly to raise an error if you'd like, it's just slightly more wordy.

```python
# this will make errors available after the context in `vtk_errors`
vtk_errors = pyvista.VtkErrorCatcher()
with vtk_errors:
    generate_vtk_error()

# this will just raise the errors
with pyvista.VtkErrorCatcher(raise_errors=True):
    generate_vtk_error()

# this is the shorthand way, which we could consider removing
with pyvista.raise_vtk_errors():
    generate_vtk_error()
```

If you guys prefer, we can either:
1. Remove `raise_vtk_error` entirely
2. Remove `raise_vtk_error` from the "public"-interface by not importing it in `pyvista.utilities`

Reply:
I vote for number 1.

Reply:
Agreed with @MatthewFlamm.  From `import this`:

> There should be one-- and preferably only one --obvious way to do it.